google/cloud/bigtable/client.py

```diff
@@ -62,29 +80,227 @@ def __init__(
                 Client options used to set user options
                 on the client. API Endpoint should be set through client_options.
             metadata: a list of metadata headers to be attached to all calls with this client
+        Raises:
+          - RuntimeError if called outside of an async run loop context
+          - ValueError if pool_size is less than 1
         """
-        raise NotImplementedError
+        # set up transport in registry
+        transport_str = f"pooled_grpc_asyncio_{pool_size}"
+        transport = PooledBigtableGrpcAsyncIOTransport.with_fixed_size(pool_size)
+        BigtableClientMeta._transport_registry[transport_str] = transport
```

Comment:
Please avoid making the pool fixed size. It can have negative effects on customers with fluctuating traffic:
- if the pool size is too small then when customers spike in qps, RPCs will get queued in grpc client
- if pool size is too big when there isnt a lot of traffic, gRPC channels will get idled and then will cause latency with every connection picked

Reply:
Do we have any examples of libraries that implement dynamic pooling like you're thinking? What sort of algorithm do you have in mind?

From what I saw, it seems the other libraries generally use fixed pool sizes, so I was hoping to keep it simple for now, and then revisit dynamic sizing after the core client is complete

Reply:
You can take a look at the java implementation: https://github.com/googleapis/gapic-generator-java/blob/main/gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/ChannelPool.java#L248

However I think it's ok to have a fix sized pool to start with, just make it extensible so that in the future we can make it dynamic?